client/src/boot/BootRoutes.js

```diff
@@ -140,8 +158,79 @@ class BootRoutes {
       });
     });
 
+    // Legacy support – make sure we don't overwrite the `onbeforeunload`
+    // handler in `LeftAndMain.EditForm.js` but run it after the new handler.
+    const currBeforeUnload = window.onbeforeunload;
+    window.onbeforeunload = () => {
+      if (this.shouldConfirmBeforeUnload()) {
```

Comment:
a bit of code duplication here: e.g. `let message = this.handleBeforeUnload(); if (message) { return message; }`